app.rb

```diff
@@ -1,83 +1,116 @@
+  end
 
-unless is_http?
-  monit_source_file = File.join(__dir__, monit_source)
+  get '/monit_status' do
+    (read_mmonit_status + read_monit_xml).to_json
+  end
 
-  raise "File does not exist!" unless File.exists?(monit_source_file)
-end
+  def read_jenkins_status
+    jenkins_adapter = JenkinsAdapter.new(ENV.fetch("JENKINS_USER"), ENV.fetch("JENKINS_TOKEN"))
 
-get '/' do
-  renderer = ERB.new(File.read('views/index.html.erb'))
-  renderer.result(binding)
-end
+    jenkins_adapter.all_builds.to_json
+  end
 
-get '/jenkins_status' do
-  read_jenkins_status
-end
+  def read_mmonit_status
+    if App.servers
+      App.servers.map do |server|
+        status = server.ok? ? 'success' : 'error'
+        {
+            name: server.name,
+            status: status
+        }
+      end
+    else
+      []
+    end
+  end
 
-get '/monit_status' do
-  read_monit_xml
-end
+  def read_monit_xml
+    xml = get_from_http
 
-def read_monit_xml
-  xml = if is_http?
-    get_from_http
-  else
-    monit_source_file = File.join(__dir__, monit_source)
+    @doc = Nokogiri::XML(xml)
 
-    File.read(monit_source_file)
+    @doc.xpath('//service').map do |service_element|
+      status = service_element.xpath('status').text == '0' ? 'success' : 'error'
+      {
+          name: service_element.xpath('name').text,
+          status: status
+      }
+    end
   end
 
-  @doc = Nokogiri::XML(xml)
+  private
 
-  @doc.xpath('//service').map do |service_element|
-    status = service_element.xpath('status').text == '0' ? 'success' : 'error'
-    {
-      name: service_element.xpath('name').text,
-      status: status
-    }
-  end.to_json
-end
+  def monit_source
+    ENV.fetch("MONIT_XML_URL")
+  end
 
-def read_jenkins_status
-  jenkins_adapter = JenkinsAdapter.new(ENV.fetch("JENKINS_USER"), ENV.fetch("JENKINS_TOKEN"))
+  def get_from_http
+    uri = URI(monit_source)
+
+    req = Net::HTTP::Get.new(uri)
+    req.basic_auth(*uri.userinfo.split(':')) if uri.userinfo
+
+    res = Net::HTTP.start(uri.hostname, uri.port) do |http|
+      http.request(req)
+    end
+
+    res.body
+  end
 
-  jenkins_adapter.all_builds.to_json
-end
 
-def get_from_http
-  uri = URI(monit_source)
+  def self.scheduler
+    @scheduler ||= Rufus::Scheduler.new
+  end
+
+  def self.load_servers
+    mmonit_adapter = MMonitAdapter.new(ENV.fetch('MMONIT_USERNAME'), ENV.fetch('MMONIT_PASSWORD'),
+                                       ENV.fetch('MMONIT_HOST'), ENV.fetch('MMONIT_PORT')
+    )
+    mmonit_adapter.connect
+    @servers = mmonit_adapter.statuses
+    mmonit_adapter.disconnect
+  end
 
-  req = Net::HTTP::Get.new(uri)
-  req.basic_auth(*uri.userinfo.split(':')) if uri.userinfo
+  def self.servers
+    @servers
+  end
+
+  configure do
+    set :bind, '0.0.0.0'
+    set :port, 9292
+    set :scheduler, App.scheduler
 
-  res = Net::HTTP.start(uri.hostname, uri.port) do |http|
-    http.request(req)
+    settings.scheduler.every '5s', first: :now do
+      App.load_servers
```

Comment:
Voor de zekerheid: Weet je zeker dat `@servers` persisted is tussen requests? Zo nee, dan zou je misschien beter bijv. een simpele cache in Redis kunnen doen:
```ruby
redis.hsetall("mmonit_server_statuses", statuses)
redis.expire("mmonit_server_statuses", 5)
```

en die laten uitlezen of instellen door de webapp. Vraag is of het nodig is: Is mmonit zo langzaam dat periodiek ophalen nodig is of kan het ook tijdens een request gedaan worden?